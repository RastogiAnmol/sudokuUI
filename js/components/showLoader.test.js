import React from "react";
import ShowLoaderComponent from "./showLoader";
import Adapter from "enzyme-adapter-react-14";
import { shallow, mount, render, configure } from "enzyme";
configure({ adapter: new Adapter() });
describe("ShowLoaderComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ShowLoaderComponent debug />);
    expect(component).toMatchSnapshot();
  });
});
