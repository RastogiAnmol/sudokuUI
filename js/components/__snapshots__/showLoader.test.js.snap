// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowLoaderComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowLoader
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src="../../images/loader.svg"
        />,
        <span
          className="dontDisplay"
        >
          2
        </span>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "../../images/loader.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 2,
          "className": "dontDisplay",
        },
        "ref": null,
        "rendered": 2,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src="../../images/loader.svg"
          />,
          <span
            className="dontDisplay"
          >
            2
          </span>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "../../images/loader.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "dontDisplay",
          },
          "ref": null,
          "rendered": 2,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
